src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -1,12 +1,70 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
 import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.FreeTextSegment;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReportStatusCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.PatientClinicalInfo;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class ProcedureRequestMapper {
-    public ProcedureRequest mapToProcedureRequest(final Message message) {
-        return new ProcedureRequest();
+    private final UUIDGenerator uuidGenerator;
+
+    public Optional<ProcedureRequest> mapToProcedureRequest(final Message message) {
+        return message.getPatientClinicalInfo().stream()
+            .map(patientClinicalInfo -> mapPatientClinicalInfo(patientClinicalInfo))
+            .findAny();
+    }
+
+    private ProcedureRequest mapPatientClinicalInfo(final PatientClinicalInfo patientClinicalInfo) {
+        final ProcedureRequest procedureRequest = new ProcedureRequest();
+
+        mapFreeText(patientClinicalInfo, procedureRequest);
+        mapStatus(patientClinicalInfo, procedureRequest);
+        procedureRequest.setIntent(ProcedureRequest.ProcedureRequestIntent.NULL);
+        procedureRequest.setCode(new CodeableConcept().setText("unknown"));
+        procedureRequest.setId(uuidGenerator.generateUUID());
+
+        return procedureRequest;
+    }
+
+    private void mapFreeText(final PatientClinicalInfo patientClinicalInfo, final ProcedureRequest procedureRequest) {
+        final ArrayList<Annotation> annotations = new ArrayList<>();
+
+        var textualClinicalInformationList = patientClinicalInfo.getFreeTexts()
+            .stream()
+            .map(FreeTextSegment::getTexts)
+            .collect(Collectors.toList());
+
+        for (var freeTextItem: textualClinicalInformationList) {
+            Annotation annotation = new Annotation();
+            annotation.setText(freeTextItem[0]);
```

Comment:
Maybe all of the texts need to be joined into one rather than just taking the first and ignoring the others?
(I haven't checked the spec to find out for sure.)

Reply:
I think that in this case, this is how the getTexts works in FreeTextSegment because it returns the String array for each FTX+CID segment. It splits the segment value by colon. I didn't find any notes that the SG10 Free text can be splitted by the colon that's why I thought that I can assume that there always will be no colons. But I think that you are right and my assumption wasn't perfect because I guess that colon still might appear in SG10 Free Text as I suppose

Reply:
Maybe this will do the work a bit better: 1d1f9a80645157c921063b3e7e634b4e00b82c77

However, I have checked some other FreeText segments and still not sure if we are supposed to split by colon all the FreeText segments or not. @matthewdoreilly maybe you have an answer? If for example, we have something like this: `"FTX+CID+++part1:part2"`, should we have one Annotation or two inside the ProcedureRequest `note` field?